lib/utilities/async_button.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:flutter/material.dart';
+
+class AsyncButton extends StatefulWidget {
+  final void Function() onPressedFunction;
+  final String label;
+
+  const AsyncButton({required this.onPressedFunction, required this.label, Key? key})
+      : super(key: key);
+
+  @override
+  State<AsyncButton> createState() => _AsyncButtonState();
+}
+
+class _AsyncButtonState extends State<AsyncButton> {
+  bool submitting = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton.icon(
+        label: Text(widget.label),
+        icon: !submitting
+            ? Container()
+            : Container(
+                width: 18,
+                height: 18,
+                padding: const EdgeInsets.all(2.0),
+                child: const CircularProgressIndicator(
+                  color: Colors.white,
+                  strokeWidth: 3,
+                ),
+              ),
+        onPressed: submitting
+            ? null
+            : () {
+                setState(() {
+                  submitting = true;
+                });
+
+                widget.onPressedFunction();
```

Comment:
🤔 Isn't it kind of risky keeping the error handling outside, as we would not be able to set the submitting state to false again if the calling widget does not properly handle the error.

I suggest we wrap that into a try catch and re-throw the exception when the action fails. This way we can add our error handling.